### PR TITLE
docs: Add test and docs for schedule `env` key

### LINKED ISSUE #6387: feat: add support for `env` key for `schedules`
Related to:

* https://github.com/meltano/meltano/issues/6386

```yaml
schedules:
- name: gitlab
  interval: '@daily'
  job: gitlab
  env:
    DBT_MODELS: +gitlab+
```

Precedence discussion in https://github.com/meltano/meltano/issues/6386#issuecomment-1179138398

### PR DESCRIPTION
**Pull request by cjohnhanson**
Closes #6387

### CODE REVIEW

File: docs/src/_guide/orchestration.md [hunk 1]
```diff
@@ -26,6 +26,26 @@ meltano job add tap-gitlab-to-target-postgres-with-dbt --tasks "tap-gitlab targe
 meltano schedule add daily-gitlab-load --job tap-gitlab-to-target-postgres-with-dbt --interval '@daily'
 ```
```
Comment (tayloramurphy) [inline docs/src/_guide/orchestration.md:28]: @cjohnhanson this text looks good. We'll need an update to https://docs.meltano.com/guide/configuration#specifying-environment-variables as well in the code and precedence blocks as well.

**Review by tayloramurphy — CHANGES_REQUESTED** [review]

File: docs/src/_guide/orchestration.md (hunk 1 above)
Comment (cjohnhanson) [reply docs/src/_guide/orchestration.md:28]: @tayloramurphy done.

The docs I added outline the _current_ behavior. Note that it turns out schedule invocation is fairly complex and correcting behavior will need to be downstream of a significant refactor in how we manage environment variables. I've created [this follow-up issue](https://github.com/meltano/meltano/issues/6675) with details.

**Review by edgarrmondragon — APPROVED** [review]

File: docs/src/_guide/configuration.md [hunk 2]
```diff
@@ -113,6 +126,7 @@ Environment levels within `meltano.yml` resolve in order of precedence (within a
 - environment level env
 - root level plugin env
 - root level env
+- schedule level env
 - .env file
 - terminal env # lowest
 ```
```
Comment (tayloramurphy) [inline docs/src/_guide/configuration.md:132]: @cjohnhanson It won't let me comment on the line I want, but will you add this after the last sentence:

```html
        <div class="notification is-info">
          <p>We are considering adding support for the <code>env</code> key to <a href="/concepts/project#jobs">jobs</a> as well as updating the precedence order when we do. We'd love to hear your thoughts in the <a href="https://github.com/meltano/meltano/issues/6386">GitHub issue</a> about this possible change!</p>
        </div>
```

**Review by tayloramurphy — CHANGES_REQUESTED** [review]

File: docs/src/_guide/configuration.md (hunk 2 above)
Comment (cjohnhanson) [reply docs/src/_guide/configuration.md:132]: 👍 Good call, added.

**Review by tayloramurphy — APPROVED** [review]

**Pull request merged**